index.js

```diff
@@ -1,38 +1,40 @@
+"use strict";
 /**
- * If the middleware function does declare receiving the `next` callback
- * assume that it's synchronous and invoke `next` ourselves
+ * If the middleware function does not declare receiving the `next` callback
+ * assume that it's synchronous and invoke `next` ourselves.
  */
 function noCallbackHandler(ctx, connectMiddleware, next) {
-  connectMiddleware(ctx.req, ctx.res)
-  return next()
+    connectMiddleware(ctx.req, ctx.res);
+    return next();
 }
-
 /**
  * The middleware function does include the `next` callback so only resolve
  * the Promise when it's called. If it's never called, the middleware stack
- * completion will stall
+ * completion will stall.
  */
 function withCallbackHandler(ctx, connectMiddleware, next) {
-  return new Promise((resolve, reject) => {
-    connectMiddleware(ctx.req, ctx.res, err => {
-      if (err) reject(err)
-      else resolve(next())
-    })
-  })
+    return new Promise((resolve, reject) => {
+        connectMiddleware(ctx.req, ctx.res, (err) => {
+            if (err)
+                reject(err);
+            else
+                resolve(next());
+        });
+    });
+}
+function hasNoCallback(middleware) {
+    return middleware.length < 3;
 }
-
 /**
  * Returns a Koa middleware function that varies its async logic based on if the
  * given middleware function declares at least 3 parameters, i.e. includes
- * the `next` callback function
+ * the `next` callback function.
  */
 function koaConnect(connectMiddleware) {
-  const handler = connectMiddleware.length < 3
-    ? noCallbackHandler
-    : withCallbackHandler
-  return function koaConnect(ctx, next) {
-    return handler(ctx, connectMiddleware, next)
-  }
+    return function koaConnect(ctx, next) {
+        return hasNoCallback(connectMiddleware)
```

Comment:
It seems the old version evaluated which handler to use once outside the returned function and stored it in a constant used in the returned function. However, if I am reading the new version correctly, it evaluates which handler to use inside the returned function. This means the condition for which handler to use will run each time the returned function is called which may result in slightly decreased performance.

Reply:
Yeah, that's a great point. Made some changes to keep the old behavior.